components/prize_pool/commons/starcraft_starcraft2/prize_pool_legacy_starcraft.lua

```diff
@@ -0,0 +1,434 @@
+	else
+		-- Otherwise handle it like it would been a points input using different parameter
+		StarcraftLegacyPrizePool._assignType(newArgs, header.localcurrency, 'localprize')
+	end
+
+	StarcraftLegacyPrizePool._assignType(newArgs, header.points, 'points')
+	StarcraftLegacyPrizePool._assignType(newArgs, header['2points'] or header.points2, 'points2')
+
+	local defaultOpponentType = Opponent.readType(header.opponentType or Opponent.solo)
+	if defaultOpponentType then
+		newArgs.type = {type = defaultOpponentType}
+	end
+	assert(newArgs.type, 'Invalid opponentType set.')
+	newArgs.type.isarchon = header.defaultIsArchon
+
+	CACHED_DATA.defaultOpponentType = defaultOpponentType
+	CACHED_DATA.defaultIsArchon = header.defaultIsArchon
+
+	newArgs.storesmw = header.storeSmw or header['smw mute']
+	newArgs.storelpdb = header.storeLpdb
+	newArgs.storeTournament = header.storeTournament
+	newArgs.series = header.series
+	newArgs.tier = header.tier
+	newArgs['tournament name'] = header['tournament name']
+
+	-- import settings
+	if header.lpdb and header.lpdb ~= 'auto' then
+		newArgs.import = header.lpdb
+	end
+	newArgs.import = StarcraftLegacyPrizePool._enableImport(newArgs)
+
+	newArgs.importLimit = header.importLimit
+	newArgs.tournament1 = header.tournament1 or header.tournament1
+	for key, tournament in Table.iter.pairsByPrefix(header, 'tournament') do
+		newArgs[key] = tournament
+	end
+	newArgs.matchGroupId1 = header.matchGroupId1 or header.matchGroupId
+	for key, matchGroupId in Table.iter.pairsByPrefix(header, 'matchGroupId') do
+		newArgs[key] = matchGroupId
+	end
+
+	local newSlotIndex = 0
+	local currentPlace
+	for _, slot in ipairs(slots) do
+		-- retrieve the slot and push it into a temp var so it can be altered (to merge slots if need be)
+		local tempSlot = StarcraftLegacyPrizePool._mapSlot(slot)
+		local place = tempSlot.place or slot.place
+		-- if we want to merge slots and the slot we just retrieved
+		-- has the same place as the one before, then append the opponents
+		if place and currentPlace == place then
+			Array.appendWith(newArgs[newSlotIndex].opponents, unpack(tempSlot.opponents))
+		else -- regular case we do not need to merge
+			currentPlace = place
+			newSlotIndex = newSlotIndex + 1
+			newArgs[newSlotIndex] = tempSlot
+		end
+	end
+
+	-- iterate over slots and merge opponents into the slots directly
+	local numberOfSlots = newSlotIndex
+	for slotIndex = 1, numberOfSlots do
+		-- if we have to merge we need to kick empty opponents
+		-- while if we import we want tom keep them
+		if
+			not newArgs.import
+			and #newArgs[slotIndex].opponents > newArgs[slotIndex].opponentsInSlot
+		then
+			newArgs[slotIndex].opponents = Array.filter(
+				newArgs[slotIndex].opponents,
+				function(opponent) return not opponent.isEmpty end
+			)
+		end
+		Table.mergeInto(newArgs[slotIndex], newArgs[slotIndex].opponents)
+		newArgs[slotIndex].opponents = nil
+	end
+
+	StarcraftLegacyPrizePool._sortQualifiers(newArgs)
```

Comment:
Skip the Qualifier Sort unless you really want it. Added it on CS request.

Reply:
will kick